### ISSUE TITLE
let remappings `zx,za...` conserve prior maps

### ISSUE THREAD

**Issue by Konfekt** [Konfekt/FastFold#14]
This could be achieved by something in the vein of
```vim
let nzx = maparg('zx','n',0,1)
if exists('nzx.rhs') | execute 'nnoremap ' . (nzx.expr ? '<expr>' : '') . ' <SID>zx '.nzx.rhs | else | nnoremap  <SID>zx zx | endif
nmap <expr> zx <SID>zx<plug>(FastFoldUpdate)
```

**Comment by kopischke** [comment]
Having done a bit of that in my personal configuration, and having taken a look at some plug-in code that tries to do that, I can tell you it is far from being this straightforward. You are in for a world of hurtful edge and corner cases :).

**Comment by Konfekt** [comment]
Yep, It'll only work for most common mappings. There are for example parameters such as `<nowait>`  missing. There is the savemap plugin, but implementing missing features of Vim in Vimscript is an exercise in futility. It'll probably have to stay as is, either users have not overwritten the fold mappings or have to implement FastFoldUpdate themselves.
